jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt

```diff
@@ -76,9 +76,7 @@ class CodeWhispererService {
         val project = editor.project ?: return
         if (!isCodeWhispererEnabled(project)) return
         if (isConnectionExpired(project)) {
-            if (triggerTypeInfo.triggerType == CodewhispererTriggerType.OnDemand) {
-                promptReAuth(project)
-            }
+            promptReAuth(project)
```

Comment:
What's the behavior if the user doesn't interact with the notification but keeps typing anyway?

Reply:
yea, since once the notification is shown once, it will logout the stale connection right away, user will be unable to trigger CW service from them on

Reply:
`promptReAuth` will try
- refresh the connection
  - if succeed -> nothing will happen
  - if fail -> show the notification telling users the connection is expired and logout from it

Reply:
and users can click on the `autehnticate` button on the notification mentioned above which will start the browser flow immediately instead of starting from the login dialog

Reply:
I see, this is dropping the persistence entirely. The new behavior might not play nicely with the UX of other features e.g. CodeCatalyst. I don't see too much risk with this change though I think we should check if the CodeCatalyst implementation has any logic to prompt on expiration.

Reply:
make sense, i'm also wondering where does CAWS require the connection/credential/access token (e.g. backend API call or something etc.)

Reply:
btw, just played around with such case (modified AWS ID `expiredAt` `refreshToken`), it will logout from both CAWS + CWSPR

Reply:
but worth verifying this with @rli

Reply:
what i found is this https://github.com/aws/aws-toolkit-jetbrains/blob/main/jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/sono/SonoCredentialManager.kt#L46

Reply:
^^ https://github.com/aws/aws-toolkit-jetbrains/blob/main/jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/sono/SonoCredentialManager.kt#L96 seems do the same thing: start the browser flow directly

Reply:
i can change the CWSPR part as same as CAWS looks like

Reply:
As long as the CWSPR change doesn't break the CodeCatalyst behavior then it's fine as-is.

Reply:
ack, will hold this PR and see if @rli is back tomorrow and if he have any comment on this thanks Jaden